src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -484,4 +500,49 @@ private function walkRegexAst(
 		}
 	}
 
+	private function getPatternType(Expr $patternExpr, Scope $scope): Type
+	{
+		if ($patternExpr instanceof Expr\BinaryOp\Concat) {
+			return $this->resolvePatternConcat($patternExpr, $scope);
+		}
+
+		return $scope->getType($patternExpr);
+	}
+
+	private function resolvePatternConcat(Expr\BinaryOp\Concat $concat, Scope $scope): Type
+	{
+		if (
+			$concat->left instanceof Expr\FuncCall
+			&& $concat->left->name instanceof Name
+			&& $concat->left->name->toLowerString() === 'preg_quote'
+		) {
+			$left = new ConstantStringType('');
```

Comment:
Is there any way to make this a non empty string if we know the preg quote is called with a non empty string?

Reply:
Yeah thats doable. I left it out because atm we don't differentiate different string types. Its a long way to go and I want to look into preg_match_all before doing more precise types